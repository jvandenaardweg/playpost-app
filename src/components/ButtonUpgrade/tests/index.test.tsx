import React from 'react';
import { fireEvent, render, RenderAPI } from 'react-native-testing-library';

import { ButtonUpgrade } from '../index';

describe('ButtonUpgrade', () => {

  describe('rendering', () => {
    let wrapper: RenderAPI;
    const onPressHandler = jest.fn();

    beforeAll(() => {
      wrapper = render(<ButtonUpgrade onPress={onPressHandler} />);
    });

    it('should render correctly', () => {
      expect(wrapper.toJSON()).toMatchSnapshot();
    });

    it('should fire onPress when the button is pressed', () => {
      fireEvent.press(wrapper.getByTestId('ButtonTiny-Button'));
      expect(onPressHandler).toHaveBeenCalledTimes(1);
    });
  });
});
