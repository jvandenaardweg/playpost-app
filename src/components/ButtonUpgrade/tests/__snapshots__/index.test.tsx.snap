// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonUpgrade rendering should render correctly 1`] = `
<View
  style={
    Object {
      "paddingRight": 12,
    }
  }
>
  <View
    accessible={true}
    clickable={true}
    isTVSelectable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "borderRadius": 28,
          "height": 28,
        },
        Object {
          "backgroundColor": "#037DE2",
        },
      ]
    }
    testID="ButtonTiny-Button"
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderRadius": 28,
            "flexDirection": "row",
            "height": 28,
            "paddingLeft": 12,
            "paddingRight": 12,
          },
          Object {
            "backgroundColor": "#037DE2",
          },
        ]
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontSize": 13,
              "fontWeight": "600",
            },
            Object {
              "color": "#ffffff",
            },
          ]
        }
      >
        Upgrade
      </Text>
      <View
        style={
          Object {
            "marginLeft": 6,
            "marginRight": -1,
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "#ffffff",
                "fontSize": 12,
              },
              undefined,
              Object {
                "fontFamily": "FontAwesome5Free-Solid",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {
                "fontWeight": "900",
              },
            ]
          }
        >
          ï€…
        </Text>
      </View>
    </View>
  </View>
</View>
`;
